lusidtools/cocoon/utilities.py

```diff
@@ -260,27 +260,40 @@ def set_attributes_recursive(
     # For each of the attributes to populate
     for key in list(populate_attributes):
 
+        # Get the attribute type
+        attribute_type = obj_attr[key]
+
         # If it is an additional attribute, populate it with the provided values and move to the next attribute
         if key in list(additional_attributes.keys()):
-            obj_init_values[key] = additional_attributes[key]
-            continue
+            # Handle identifiers provided within instrument definition (e.g. 'Bond', 'Future', etc.)
+            if (key, attribute_type) == ("identifiers", "dict(str, str)"):
+                obj_init_values[key] = {
+                    str_key: row[str_value["value"]]
+                    for str_key, str_value in mapping[key].items()
+                }
+                continue
+            else:
+                obj_init_values[key] = additional_attributes[key]
+                continue
 
         # This block keeps track of the number of missing (non-additional) attributes
         else:
             total_count += 1
             if mapping[key] is None:
                 none_count += 1
 
-        # Get the attribute type
-        attribute_type = obj_attr[key]
-
         # If this is the last object and there is no more nesting set the value from the row
         if not isinstance(mapping[key], dict):
             # If this exists in the mapping with a value and there is a value in the row for it
             if mapping[key] is not None and not pd.isna(row[mapping[key]]):
                 # Converts to a date if it is a date field
                 if "date" in key or "created" in key or "effective_at" in key:
                     obj_init_values[key] = str(DateOrCutLabel(row[mapping[key]]))
+                # Converts to a list element if it is a list field
+                elif "list" in attribute_type and not isinstance(
+                    row[mapping[key]], list
+                ):
+                    obj_init_values[key] = list(row[mapping[key]].split())
```

Comment:
Does `obj_init_values[key] = [row[mapping[key]]]` work?  That is what we use on line 318 below and it is simpler than splitting.